go/adbc/driver/flightsql/flightsql_statement.go

```diff
@@ -132,6 +153,9 @@ type statement struct {
 	prepared  *flightsql.PreparedStatement
 	queueSize int
 	timeouts  timeoutOption
+	// TODO: this needs to be cleared at appropriate times
```

Comment:
should we create a corresponding issue to make sure we don't forget to do this?

Reply:
Ah, I meant to do this before opening it, thanks. I've updated it so that entrypoints reset the incremental execution state where appropriate.